utils/src/ooinstall/install_transactions.py

```diff
@@ -41,6 +44,10 @@ def generate_inventory(hosts):
     if 'OO_INSTALL_STAGE_REGISTRY' in os.environ:
         base_inventory.write('oreg_url=registry.access.stage.redhat.com/openshift3/ose-${component}:${version}\n')
 
+    if any(host.hostname == installer_host for host in hosts):
+        base_inventory.write("ansible_connection=local\n")
+        base_inventory.write("ansible_sudo=no\n")
+
```

Comment:
We should also add a check to verify that the user is running as root.

If they are not running as root, we can run `sudo -v --non-interactive` to test if they have password-less sudo enabled, which will return an error code of 1 if a password is required or if the user does not have sudo permissions (the error message will have to be parsed to tell which). If the command is successful we can set ansible_sudo=yes and invoke ansible-playbook normally.

On a side note, we should probably switch to using ansible_become instead of ansible_sudo